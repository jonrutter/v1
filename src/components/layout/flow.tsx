import React from 'react';
import clsx from 'clsx';

type Props<T extends React.ElementType> = {
  as?: T;
  className?: string;
  size?: number;
};

/**
 * Polymorphic component providing standardized vertical spacing between a group of children
 */
export const Flow = <T extends React.ElementType = 'div'>({
  as,
  className,
  children,
}: Props<T> & Omit<React.ComponentPropsWithoutRef<T>, keyof Props<T>>) => {
  const Tag = as || 'div';

  return (
    <Tag className={clsx('space-y-8 md:space-y-10 lg:space-y-14', className)}>
      {children}
    </Tag>
  );
};
