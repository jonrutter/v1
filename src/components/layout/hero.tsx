import React from 'react';
import clsx from 'clsx';

type Props = {
  short?: boolean;
};

/**
 * Renders a page's hero section.
 */
export const Hero: React.FC<Props> = ({ children, short }) => (
  <div
    className={clsx(
      'py-12 px-6 md:py-24 md:px-12 transition-all lg:flex lg:justify-center lg:items-center bg-white dark:bg-slate-900 text-base md:text-lg',
      !short && 'lg:min-h-[calc(100vh-5rem)]'
    )}
  >
    <div className="mx-auto max-w-prose">{children}</div>
  </div>
);
