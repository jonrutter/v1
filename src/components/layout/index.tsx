import React from 'react';

// components
import { Footer } from './footer';
import { Nav } from './nav';
import { SkipToContent } from './skip-to-main';

type Props = {
  withCTA?: boolean;
};

/**
 * Main layout component for all pages on the site.
 *
 * @param props.withCTA - (Optional) renders a Call to action box at the bottom of the page.
 */
export const Layout: React.FC<Props> = ({ withCTA, children }) => (
  <div className="overflow-x-hidden max-w-[100vw] bg-white dark:bg-slate-900 transition-all text-slate-700 dark:text-slate-200">
    <SkipToContent />
    <Nav />
    <div id="main">
      <main>{children}</main>
      <Footer short={!withCTA} />
    </div>
  </div>
);

export * from './footer';
export * from './nav';
export * from './hero';
export * from './section';
export * from './skip-to-main';
