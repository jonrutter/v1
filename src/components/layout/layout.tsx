import React from 'react';

// components
import { Footer } from './footer';
import { Nav } from './nav';
import { SkipToContent } from './skip-to-main';

type Props = {
  blogPost?: boolean;
};

/**
 * Main layout component for all pages on the site.
 */
export const Layout: React.FC<Props> = ({ blogPost, children }) => (
  <div
    className={`overflow-x-hidden max-w-[100vw] text-slate-600 dark:text-slate-200 min-h-screen ${
      blogPost ? 'bg-white dark:bg-slate-900' : 'bg-white dark:bg-slate-800'
    }`}
  >
    <SkipToContent />
    <Nav />
    <div id="main">
      <main>{children}</main>
      <Footer blogPost={!!blogPost} />
    </div>
  </div>
);
