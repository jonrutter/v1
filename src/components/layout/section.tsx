import React from 'react';
import clsx from 'clsx';

// types
import type { PolymorphicProps } from '@/utils/polymorphic';

type Props = {
  className?: string;
};

/**
 * Renders a block of content, styled with the site's default spacing for consistency.
 *
 * Should be used as a direct child of `<Layout>`, in order to maintain correct site layout.
 */
export const Section = <T extends React.ElementType = 'section'>({
  as,
  className = '',
  children,
}: PolymorphicProps<T, Props>) => {
  const Tag = as || 'section';
  return (
    <div
      className={clsx(
        'py-12 sm:py-16 md:py-20 px-6 md:px-12 transition-all bg-inherit text-inherit',
        className
      )}
    >
      <Tag className="max-w-site-sm md:max-w-site-md lg:max-w-site-lg mx-auto">
        {children}
      </Tag>
    </div>
  );
};
