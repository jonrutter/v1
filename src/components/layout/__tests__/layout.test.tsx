import React from 'react';
import { render, screen } from 'test-utils';

import { Layout } from '@/components/layout';
import { PageWrap } from '@/components/app';

const Component = (
  <PageWrap>
    <Layout>
      <h1>Hello, world!</h1>
    </Layout>
  </PageWrap>
);

const BlogComponent = (
  <PageWrap>
    <Layout blogPost>
      <h1>Hello, world!</h1>
    </Layout>
  </PageWrap>
);

describe('Layout', () => {
  it('renders the correct content', () => {
    render(Component);

    // page content should be rendered
    screen.getByText('Hello, world!');
  });
  it('renders correctly as a blog post', () => {
    render(BlogComponent);

    // page content should be rendered
    screen.getByText('Hello, world!');
  });
});
