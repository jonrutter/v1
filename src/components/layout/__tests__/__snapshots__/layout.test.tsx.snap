// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout matches the snapshot 1`] = `
<div
  className="overflow-x-hidden max-w-[100vw] bg-white dark:bg-slate-900 transition-all text-slate-700 dark:text-slate-200"
>
  <a
    className="fixed top-24 left-0 bg-white text-slate-900 dark:bg-slate-900 dark:text-slate-50 z-[9999] text-lg py-3 px-6 underline -translate-x-full focus:translate-x-1 transition-all rounded-md shadow-md outline-none focus:ring-2 focus:ring-slate-900 focus:dark:ring-slate-50"
    href="#main"
  >
    Skip to content
  </a>
  <nav
    className="py-8 lg:py-12 px-6 md:px-12 w-full z-40 transition-all text-slate-900 dark:text-slate-50 motion-reduce:!translate-y-0 bg-transparent"
  >
    <div
      className="max-w-site-full mx-auto flex justify-between items-center"
    >
      <div
        className="max-w-[3rem] h-auto"
      >
        <svg
          aria-label="Jon Rutter"
          className="w-full h-auto"
          height="512"
          role="img"
          version="1.1"
          viewBox="0 0 135.47 135.47"
          width="512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            transform="translate(-123.5 -173.39)"
          >
            <circle
              cx="191.23"
              cy="241.12"
              fill="#0f172a"
              r="65.006"
              stroke="#fff"
              strokeWidth="5.454"
            />
            <g
              fill="#fff"
              transform="matrix(2.5686 0 0 2.5686 58.78 88.635)"
            >
              <path
                d="m48.569 50.77h-4.1411v25.299h4.1411z"
              />
              <path
                d="m61.506 57.907c1.9678 0 3.5629-1.5928 3.5629-3.5578 0-1.9649-1.5951-3.5577-3.5629-3.5577-1.9677 0-3.5629 1.5928-3.5629 3.5577 0 1.965 1.5952 3.5578 3.5629 3.5578z"
              />
              <path
                d="m57.939 50.77h-4.1411v16.602h4.1411z"
              />
              <path
                d="m46.519 46.101c1.1358 0 2.0566-0.9193 2.0566-2.0534 0-1.134-0.9208-2.0533-2.0566-2.0533-1.1357 0-2.0564 0.9193-2.0564 2.0533 0 1.1341 0.9207 2.0534 2.0564 2.0534z"
              />
            </g>
          </g>
        </svg>
      </div>
      <ul
        className="hidden md:flex items-center space-x-8 font-heading font-medium text-lg text-slate-600 dark:text-slate-300"
      >
        <li>
          <a
            className="overflow-hidden relative transition-all before:block before:absolute before:bottom-0 before:left-1/4 before:h-[3px] before:w-1/2 before:rounded-sm before:transition-all before:bg-current before:scale-x-0 hover:before:scale-x-100 focus:before:scale-x-100 outline-none pb-1"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            className="overflow-hidden relative transition-all before:block before:absolute before:bottom-0 before:left-1/4 before:h-[3px] before:w-1/2 before:rounded-sm before:transition-all before:bg-current before:scale-x-0 hover:before:scale-x-100 focus:before:scale-x-100 outline-none pb-1"
            href="/portfolio"
          >
            Portfolio
          </a>
        </li>
        <li>
          <a
            className="overflow-hidden relative transition-all before:block before:absolute before:bottom-0 before:left-1/4 before:h-[3px] before:w-1/2 before:rounded-sm before:transition-all before:bg-current before:scale-x-0 hover:before:scale-x-100 focus:before:scale-x-100 outline-none pb-1"
            href="/blog"
          >
            Blog
          </a>
        </li>
        <li>
          <a
            className="overflow-hidden relative transition-all before:block before:absolute before:bottom-0 before:left-1/4 before:h-[3px] before:w-1/2 before:rounded-sm before:transition-all before:bg-current before:scale-x-0 hover:before:scale-x-100 focus:before:scale-x-100 outline-none pb-1"
            href="/contact"
          >
            Contact
          </a>
        </li>
      </ul>
      <ul
        className="hidden lg:flex space-x-4 items-center"
      >
        <li>
          <a
            aria-label="My GitHub"
            className="relative flex items-center justify-center group outline-none text-slate-600 dark:text-slate-300 hover:text-slate-900 dark:hover:text-slate-50 focus:text-slate-900 dark:focus:text-slate-50"
            href="https://github.com/rutterjt"
            style={
              Object {
                "fontSize": "1.25rem",
                "height": "2.5rem",
                "width": "2.5rem",
              }
            }
          >
            <svg
              aria-hidden={true}
              fill="currentColor"
              height="1em"
              role="presentation"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 496 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
              />
            </svg>
            <div
              aria-hidden={true}
              className="absolute top-0 left-0 right-0 bottom-0 rounded-full transition-all border-2 border-current opacity-0 group-hover:opacity-100 group-focus:opacity-100 outline-none overflow-hidden"
            />
          </a>
        </li>
        <li>
          <a
            aria-label="My Twitter"
            className="relative flex items-center justify-center group outline-none text-slate-600 dark:text-slate-300 hover:text-slate-900 dark:hover:text-slate-50 focus:text-slate-900 dark:focus:text-slate-50"
            href="https://twitter.com/rutterjt"
            style={
              Object {
                "fontSize": "1.25rem",
                "height": "2.5rem",
                "width": "2.5rem",
              }
            }
          >
            <svg
              aria-hidden={true}
              fill="currentColor"
              height="1em"
              role="presentation"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
              />
            </svg>
            <div
              aria-hidden={true}
              className="absolute top-0 left-0 right-0 bottom-0 rounded-full transition-all border-2 border-current opacity-0 group-hover:opacity-100 group-focus:opacity-100 outline-none overflow-hidden"
            />
          </a>
        </li>
        <li>
          <a
            aria-label="My LinkedIn"
            className="relative flex items-center justify-center group outline-none text-slate-600 dark:text-slate-300 hover:text-slate-900 dark:hover:text-slate-50 focus:text-slate-900 dark:focus:text-slate-50"
            href="https://www.linkedin.com/in/rutterjt/"
            style={
              Object {
                "fontSize": "1.25rem",
                "height": "2.5rem",
                "width": "2.5rem",
              }
            }
          >
            <svg
              aria-hidden={true}
              fill="currentColor"
              height="1em"
              role="presentation"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 0 1 107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z"
              />
            </svg>
            <div
              aria-hidden={true}
              className="absolute top-0 left-0 right-0 bottom-0 rounded-full transition-all border-2 border-current opacity-0 group-hover:opacity-100 group-focus:opacity-100 outline-none overflow-hidden"
            />
          </a>
        </li>
        <li>
          <a
            aria-label="Email me"
            className="relative flex items-center justify-center group outline-none text-slate-600 dark:text-slate-300 hover:text-slate-900 dark:hover:text-slate-50 focus:text-slate-900 dark:focus:text-slate-50"
            href="mailto:contact@jonrutter.io"
            style={
              Object {
                "fontSize": "1.25rem",
                "height": "2.5rem",
                "width": "2.5rem",
              }
            }
          >
            <svg
              aria-hidden={true}
              fill="currentColor"
              height="1em"
              role="presentation"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
              />
            </svg>
            <div
              aria-hidden={true}
              className="absolute top-0 left-0 right-0 bottom-0 rounded-full transition-all border-2 border-current opacity-0 group-hover:opacity-100 group-focus:opacity-100 outline-none overflow-hidden"
            />
          </a>
        </li>
      </ul>
      <div
        className="flex items-center space-x-4"
      >
        <div
          className="hidden md:block"
        />
        <div
          className="block lg:hidden"
        >
          <button
            aria-label="Open navigation menu"
            className="relative flex items-center justify-center group outline-none w-12 h-12 text-2xl text-slate-600 dark:text-slate-300 hover:text-slate-900 dark:hover:text-white focus:text-slate-900 dark:focus:text-white z-0"
            onClick={[Function]}
            title="Open navigation menu"
          >
            <div
              aria-hidden={true}
              className="absolute top-0 left-0 right-0 bottom-0 rounded-full transition-all-with-shadow border-2 border-slate-300 dark:border-slate-700 group-hover:border-slate-900 dark:group-hover:border-white group-focus:border-slate-900 dark:group-focus:border-white outline-none overflow-hidden"
            />
            <svg
              aria-hidden={true}
              fill="none"
              height="1em"
              role="presentation"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M2 6C2 5.44772 2.44772 5 3 5H21C21.5523 5 22 5.44772 22 6C22 6.55228 21.5523 7 21 7H3C2.44772 7 2 6.55228 2 6Z"
                fill="currentColor"
              />
              <path
                d="M2 12.0322C2 11.4799 2.44772 11.0322 3 11.0322H21C21.5523 11.0322 22 11.4799 22 12.0322C22 12.5845 21.5523 13.0322 21 13.0322H3C2.44772 13.0322 2 12.5845 2 12.0322Z"
                fill="currentColor"
              />
              <path
                d="M3 17.0645C2.44772 17.0645 2 17.5122 2 18.0645C2 18.6167 2.44772 19.0645 3 19.0645H21C21.5523 19.0645 22 18.6167 22 18.0645C22 17.5122 21.5523 17.0645 21 17.0645H3Z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </nav>
  <div
    id="main"
  >
    <main>
      <h1>
        Hello, world!
      </h1>
    </main>
    <div
      className="bg-slate-900 text-slate-50 px-6 md:px-12 pb-12 pt-28 text-lg"
    >
      <footer
        className="max-w-screen-lg mx-auto"
      >
        <div
          className="flex flex-col md:flex-row items-center text-center mb-16 md:justify-between"
        >
          <div
            className="mb-12 md:mr-8 md:mb-0 w-12 h-auto"
          >
            <svg
              aria-label="Jon Rutter"
              className="w-full h-auto"
              height="512"
              role="img"
              version="1.1"
              viewBox="0 0 135.47 135.47"
              width="512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                transform="translate(-123.5 -173.39)"
              >
                <circle
                  cx="191.23"
                  cy="241.12"
                  fill="#0f172a"
                  r="65.006"
                  stroke="#fff"
                  strokeWidth="5.454"
                />
                <g
                  fill="#fff"
                  transform="matrix(2.5686 0 0 2.5686 58.78 88.635)"
                >
                  <path
                    d="m48.569 50.77h-4.1411v25.299h4.1411z"
                  />
                  <path
                    d="m61.506 57.907c1.9678 0 3.5629-1.5928 3.5629-3.5578 0-1.9649-1.5951-3.5577-3.5629-3.5577-1.9677 0-3.5629 1.5928-3.5629 3.5577 0 1.965 1.5952 3.5578 3.5629 3.5578z"
                  />
                  <path
                    d="m57.939 50.77h-4.1411v16.602h4.1411z"
                  />
                  <path
                    d="m46.519 46.101c1.1358 0 2.0566-0.9193 2.0566-2.0534 0-1.134-0.9208-2.0533-2.0566-2.0533-1.1357 0-2.0564 0.9193-2.0564 2.0533 0 1.1341 0.9207 2.0534 2.0564 2.0534z"
                  />
                </g>
              </g>
            </svg>
          </div>
          <ul
            className="flex flex-col mb-8 md:flex-row md:items-center md:m-0 space-y-6 md:space-y-0 md:space-x-4 lg:space-x-8"
          >
            <li>
              <a
                className="overflow-hidden relative transition-all before:block before:absolute before:bottom-0 before:left-1/4 before:h-[3px] before:w-1/2 before:rounded-sm before:transition-all before:bg-current before:scale-x-0 hover:before:scale-x-100 focus:before:scale-x-100 outline-none pb-1 text-white"
                href="/"
              >
                Home
              </a>
            </li>
            <li>
              <a
                className="overflow-hidden relative transition-all before:block before:absolute before:bottom-0 before:left-1/4 before:h-[3px] before:w-1/2 before:rounded-sm before:transition-all before:bg-current before:scale-x-0 hover:before:scale-x-100 focus:before:scale-x-100 outline-none pb-1 text-white"
                href="/portfolio"
              >
                Portfolio
              </a>
            </li>
            <li>
              <a
                className="overflow-hidden relative transition-all before:block before:absolute before:bottom-0 before:left-1/4 before:h-[3px] before:w-1/2 before:rounded-sm before:transition-all before:bg-current before:scale-x-0 hover:before:scale-x-100 focus:before:scale-x-100 outline-none pb-1 text-white"
                href="/blog"
              >
                Blog
              </a>
            </li>
            <li>
              <a
                className="overflow-hidden relative transition-all before:block before:absolute before:bottom-0 before:left-1/4 before:h-[3px] before:w-1/2 before:rounded-sm before:transition-all before:bg-current before:scale-x-0 hover:before:scale-x-100 focus:before:scale-x-100 outline-none pb-1 text-white"
                href="/contact"
              >
                Contact
              </a>
            </li>
          </ul>
          <ul
            className="flex text-2xl space-x-1 lg:space-x-4"
          >
            <li>
              <a
                aria-label="My GitHub"
                className="relative flex items-center justify-center group outline-none"
                href="https://github.com/rutterjt"
                rel="noreferrer"
                style={
                  Object {
                    "fontSize": "1.5rem",
                    "height": "3rem",
                    "width": "3rem",
                  }
                }
                target="_blank"
                title="My GitHub"
              >
                <svg
                  aria-hidden={true}
                  fill="currentColor"
                  height="1em"
                  role="presentation"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 496 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                  />
                </svg>
                <div
                  aria-hidden={true}
                  className="absolute top-0 left-0 right-0 bottom-0 rounded-full transition-all border-2 border-current opacity-0 group-hover:opacity-100 group-focus:opacity-100 outline-none overflow-hidden"
                />
              </a>
            </li>
            <li>
              <a
                aria-label="My Twitter"
                className="relative flex items-center justify-center group outline-none"
                href="https://twitter.com/rutterjt"
                rel="noreferrer"
                style={
                  Object {
                    "fontSize": "1.5rem",
                    "height": "3rem",
                    "width": "3rem",
                  }
                }
                target="_blank"
                title="My Twitter"
              >
                <svg
                  aria-hidden={true}
                  fill="currentColor"
                  height="1em"
                  role="presentation"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                  />
                </svg>
                <div
                  aria-hidden={true}
                  className="absolute top-0 left-0 right-0 bottom-0 rounded-full transition-all border-2 border-current opacity-0 group-hover:opacity-100 group-focus:opacity-100 outline-none overflow-hidden"
                />
              </a>
            </li>
            <li>
              <a
                aria-label="My LinkedIn"
                className="relative flex items-center justify-center group outline-none"
                href="https://www.linkedin.com/in/rutterjt/"
                rel="noreferrer"
                style={
                  Object {
                    "fontSize": "1.5rem",
                    "height": "3rem",
                    "width": "3rem",
                  }
                }
                target="_blank"
                title="My LinkedIn"
              >
                <svg
                  aria-hidden={true}
                  fill="currentColor"
                  height="1em"
                  role="presentation"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 0 1 107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z"
                  />
                </svg>
                <div
                  aria-hidden={true}
                  className="absolute top-0 left-0 right-0 bottom-0 rounded-full transition-all border-2 border-current opacity-0 group-hover:opacity-100 group-focus:opacity-100 outline-none overflow-hidden"
                />
              </a>
            </li>
            <li>
              <a
                aria-label="Email me"
                className="relative flex items-center justify-center group outline-none"
                href="mailto:contact@jonrutter.io"
                rel="noreferrer"
                style={
                  Object {
                    "fontSize": "1.5rem",
                    "height": "3rem",
                    "width": "3rem",
                  }
                }
                target="_blank"
                title="Email me"
              >
                <svg
                  aria-hidden={true}
                  fill="currentColor"
                  height="1em"
                  role="presentation"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
                  />
                </svg>
                <div
                  aria-hidden={true}
                  className="absolute top-0 left-0 right-0 bottom-0 rounded-full transition-all border-2 border-current opacity-0 group-hover:opacity-100 group-focus:opacity-100 outline-none overflow-hidden"
                />
              </a>
            </li>
          </ul>
        </div>
        <p
          className="text-base text-center"
        >
          <a
            className="overflow-hidden relative transition-all before:block before:absolute before:bottom-0 before:left-1/4 before:h-[3px] before:w-1/2 before:rounded-sm before:transition-all before:bg-current before:scale-x-0 hover:before:scale-x-100 focus:before:scale-x-100 outline-none pb-1 text-white"
            href="https://github.com/rutterjt/v1/tree/main#license"
            rel="noreferrer"
            target="_blank"
          >
            © 
            2022
            , Jon Rutter
          </a>
        </p>
      </footer>
    </div>
  </div>
</div>
`;
